(enh) Aider-Bench: make resumable with skip_num arg

tobitege opened the pull request:
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Make Aider-Bench run resumable with optional `SKIP_NUM` env flag

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

- added optional `SKIP_NUM` env flag (int) which specifies the number of instances in the dataset to be skipped first, thus offering to resume a bench
- in `shared.py`, the `prepare_dataset` got a new optional parameter
- in that method, if a `skip_num` was given, it will combine this with the `eval_n_limit` number, so a run could skip the first 42 incidents and then run 10 instances
- temporarily changed links for huggingface evaluations to use the old repo name as the new one while the new one is in the works

xingyaoww commented on evaluation/utils/shared.py:
```diff
@@ -181,34 +181,43 @@ def prepare_dataset(
     id_column = 'instance_id'
     logger.info(f'Writing evaluation output to {output_file}')
-    finished_ids = set()
+    finished_ids: set[int] = set()
```
Actually, the ID is not strictly INT - in fact, they are mostly string across all the evals, which makes start_id harder to support?

tobitege replied:
Oops, I'll change it to str then of course.

tobitege replied:
Fixed

tobitege commented:
@xingyaoww I've switched from `start_id` to `skip_num` instead, meaning the first `skip_num` entries from the dataset are skipped.
This removes the issues related to trying to use the main id as an int completely and can still be useful.
Changed documentation accordingly.

xingyaoww commented on evaluation/README.md:
```diff
@@ -3,12 +3,14 @@
-    - For model outputs, they should be stored at [this huggingface space](https://huggingface.co/spaces/OpenHands/evaluation) for visualization.
-  - Important data files of manageable size and analysis scripts (e.g., jupyter notebooks) can be directly uploaded to this repo.
+- Raw data and experimental records should not be stored within this repo.
+- For model outputs, they should be stored at [this huggingface space](https://huggingface.co/spaces/OpenHands/evaluation) for visualization.
```
We should use OpenDevin/evaluation as link for now until i fixed it :(

tobitege replied:
Ohh, yes, that was a timely overlap now lol

tobitege replied:
Fixed in several files

xingyaoww reviewed (approved):
LGTM! Thanks!!
